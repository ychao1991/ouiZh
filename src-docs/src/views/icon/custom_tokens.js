/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiToken,
  OuiCodeBlock,
  OuiSplitPanel,
  OuiSpacer,
} from '../../../../src/components';

export default () => (
  <>
    <OuiSplitPanel.Outer hasShadow={false} hasBorder={true} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiToken iconType="tokenStruct" size="xs" color="gray" />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="none" color="subdued">
        <OuiCodeBlock
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {'<OuiToken iconType="tokenStruct" size="xs" color="gray" />'}
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>
    <OuiSpacer />
    <OuiSplitPanel.Outer hasShadow={false} hasBorder={true} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiToken iconType="tokenStruct" fill="none" />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="none" color="subdued">
        <OuiCodeBlock
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {'<OuiToken iconType="tokenStruct" fill="none" />'}
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>
    <OuiSpacer />
    <OuiSplitPanel.Outer hasShadow={false} hasBorder={true} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiToken
          iconType="tokenStruct"
          size="m"
          shape="circle"
          color="#FF0000"
        />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="none" color="subdued">
        <OuiCodeBlock
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {
            '<OuiToken iconType="tokenStruct" size="m" shape="circle" color="#FF0000" />'
          }
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>
    <OuiSpacer />
    <OuiSplitPanel.Outer hasShadow={false} hasBorder={true} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiToken
          iconType="faceNeutral"
          size="l"
          color="ouiColorVis7"
          shape="rectangle"
          fill="dark"
        />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="none" color="subdued">
        <OuiCodeBlock
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {
            '<OuiToken iconType="faceNeutral" size="l" color="ouiColorVis7" shape="rectangle" fill="dark" />'
          }
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>
  </>
);
