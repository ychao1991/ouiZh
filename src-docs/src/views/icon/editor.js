/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCopy,
} from '../../../../src/components';

const iconTypes = [
  'editorAlignCenter',
  'editorAlignLeft',
  'editorAlignRight',
  'editorBold',
  'editorCodeBlock',
  'editorComment',
  'editorDistributeHorizontal',
  'editorDistributeVertical',
  'editorHeading',
  'editorItalic',
  'editorItemAlignBottom',
  'editorItemAlignCenter',
  'editorItemAlignLeft',
  'editorItemAlignMiddle',
  'editorItemAlignRight',
  'editorItemAlignTop',
  'editorLink',
  'editorOrderedList',
  'editorPositionBottomLeft',
  'editorPositionBottomRight',
  'editorPositionTopLeft',
  'editorPositionTopRight',
  'editorRedo',
  'editorStrike',
  'editorTable',
  'editorUnderline',
  'editorUndo',
  'editorUnorderedList',
];

export default () => (
  <OuiFlexGrid direction="column" columns={3}>
    {iconTypes.map((iconType) => (
      <OuiFlexItem key={iconType}>
        <OuiCopy
          display="block"
          textToCopy={iconType}
          afterMessage={`${iconType} copied`}>
          {(copy) => (
            <OuiPanel
              hasShadow={false}
              hasBorder={false}
              onClick={copy}
              paddingSize="s">
              <OuiIcon className="oui-alignMiddle" type={iconType} /> &emsp;{' '}
              <small>{iconType}</small>
            </OuiPanel>
          )}
        </OuiCopy>
      </OuiFlexItem>
    ))}
  </OuiFlexGrid>
);
