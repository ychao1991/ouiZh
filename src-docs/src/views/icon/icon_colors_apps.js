/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import classNames from 'classnames';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiSpacer,
  OuiCodeBlock,
  OuiCopy,
} from '../../../../src/components';

const iconColors = [
  'default',
  'inherit',
  'primary',
  'success',
  'accent',
  'warning',
  'danger',
  'text',
  'subdued',
  'ghost',
  '#DA8B45',
  '#DDDDDD',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="gisApp" color="primary" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconColors.map((iconColor) => (
        <OuiFlexItem key={iconColor}>
          <OuiCopy display="block" textToCopy={`color="${iconColor}"`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                className={classNames({
                  guideDemo__ghostBackground: iconColor === 'ghost',
                })}
                paddingSize="s">
                <OuiIcon type="gisApp" color={iconColor} />
                &emsp; <small>{iconColor}</small>
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
