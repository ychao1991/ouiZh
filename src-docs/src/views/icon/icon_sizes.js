/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiCodeBlock,
  OuiSpacer,
  OuiCopy,
} from '../../../../src/components';

const iconSizes = ['s', 'm', 'l', 'xl', 'xxl', 'original'];
const iconSizesText = [
  'small',
  'medium',
  'large',
  'x-large',
  'xx-large',
  'original',
];

export default () => (
  <>
    <OuiCodeBlock language="html" isCopyable paddingSize="m">
      {'<OuiIcon type="logoElasticsearch" size="xl" />'}
    </OuiCodeBlock>
    <OuiSpacer />
    <OuiFlexGrid direction="column" columns={3}>
      {iconSizes.map((iconSize, index) => (
        <OuiFlexItem key={iconSize}>
          <OuiCopy
            display="block"
            textToCopy={iconSize}
            afterMessage={`${iconSize} copied`}>
            {(copy) => (
              <OuiPanel
                hasShadow={false}
                hasBorder={false}
                onClick={copy}
                paddingSize="s">
                <OuiIcon type="logoElasticStack" size={iconSize} /> &emsp;{' '}
                {iconSizesText[index]}
              </OuiPanel>
            )}
          </OuiCopy>
        </OuiFlexItem>
      ))}
    </OuiFlexGrid>
  </>
);
