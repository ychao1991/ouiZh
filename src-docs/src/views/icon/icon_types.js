/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGroup,
  OuiFlexItem,
  OuiIcon,
  OuiText,
  OuiSpacer,
  OuiButton,
  OuiSplitPanel,
  OuiCodeBlock,
} from '../../../../src/components';

import reactSvg from '../../images/custom.svg';

export default () => (
  <div>
    <OuiSplitPanel.Outer hasShadow={false} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiIcon
          type="https://upload.wikimedia.org/wikipedia/commons/0/02/SVG_logo.svg"
          size="xl"
          title="My SVG logo"
        />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="s" color="subdued">
        <OuiCodeBlock
          className="oui-textBreakWord"
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {
            '<OuiIcon type="https://upload.wikimedia.org/wikipedia/commons/0/02/SVG_logo.svg" size="xl" title="My SVG logo" />'
          }
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>
    <OuiSpacer />
    <OuiSplitPanel.Outer hasShadow={false} direction="row">
      <OuiSplitPanel.Inner
        className="oui-textCenter"
        grow={false}
        style={{ minWidth: 96 }}>
        <OuiIcon type={reactSvg} size="xl" title="Custom SVG icon" />
      </OuiSplitPanel.Inner>
      <OuiSplitPanel.Inner paddingSize="s" color="subdued">
        <OuiCodeBlock
          language="html"
          isCopyable
          transparentBackground
          paddingSize="m">
          {'<OuiIcon type={reactSvg} size="xl" title="Custom SVG icon" />'}
        </OuiCodeBlock>
      </OuiSplitPanel.Inner>
    </OuiSplitPanel.Outer>

    <OuiSpacer />

    <OuiText>
      <p>
        Any component that utlizes <strong>OuiIcon</strong> can use custom SVGs
        as well.
      </p>
    </OuiText>

    <OuiSpacer />

    <OuiFlexGroup>
      <OuiFlexItem grow={false}>
        <OuiButton
          iconType="https://upload.wikimedia.org/wikipedia/commons/0/02/SVG_logo.svg"
          title="Another SVG Logo">
          http://some.svg
        </OuiButton>
      </OuiFlexItem>
      <OuiFlexItem grow={false}>
        <OuiButton iconType={reactSvg}>{'{reactSvg}'}</OuiButton>
      </OuiFlexItem>
    </OuiFlexGroup>
  </div>
);
