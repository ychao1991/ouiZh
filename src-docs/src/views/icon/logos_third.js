/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import {
  OuiFlexGrid,
  OuiFlexItem,
  OuiIcon,
  OuiPanel,
  OuiText,
  OuiCopy,
} from '../../../../src/components';

const iconTypes = [
  'logoAerospike',
  'logoApache',
  'logoAWS',
  'logoAWSMono',
  'logoAzure',
  'logoAzureMono',
  'logoCeph',
  'logoCodesandbox',
  'logoCouchbase',
  'logoDocker',
  'logoDropwizard',
  'logoEtcd',
  'logoGCP',
  'logoGCPMono',
  'logoGithub',
  'logoGmail',
  'logoGolang',
  'logoGoogleG',
  'logoHAproxy',
  'logoIBM',
  'logoIBMMono',
  'logoKafka',
  'logoKubernetes',
  'logoMemcached',
  'logoMongodb',
  'logoMySQL',
  'logoNginx',
  'logoOsquery',
  'logoPhp',
  'logoPostgres',
  'logoPrometheus',
  'logoRabbitmq',
  'logoRedis',
  'logoSketch',
  'logoSlack',
  'logoWebhook',
  'logoWindows',
].sort();

export default () => (
  <OuiFlexGrid columns={4}>
    {iconTypes.map((iconType) => (
      <OuiFlexItem
        className="guideDemo__icon"
        key={iconType}
        style={{ width: '200px' }}>
        <OuiCopy textToCopy={iconType} afterMessage={`${iconType} copied`}>
          {(copy) => (
            <OuiPanel onClick={copy} className="oui-textCenter">
              <OuiIcon type={iconType} size="xl" />
              <OuiText size="s">
                <p>{iconType}</p>
              </OuiText>
            </OuiPanel>
          )}
        </OuiCopy>
      </OuiFlexItem>
    ))}
  </OuiFlexGrid>
);
