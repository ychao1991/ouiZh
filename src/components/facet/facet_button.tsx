/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, {
  FunctionComponent,
  HTMLAttributes,
  MouseEventHandler,
  ReactNode,
  RefCallback,
  ReactElement,
} from 'react';
import classNames from 'classnames';

import { CommonProps } from '../common';

import { OuiNotificationBadge } from '../badge';

import { OuiLoadingSpinner } from '../loading';
import { OuiInnerText } from '../inner_text';

export interface OuiFacetButtonProps
  extends CommonProps,
    Omit<HTMLAttributes<HTMLButtonElement>, 'onClick'> {
  buttonRef?: RefCallback<HTMLButtonElement>;
  /**
   * ReactNode to render as this component's content
   */
  children: ReactNode;
  /**
   * Any node, but preferably a `OuiIcon` or `OuiAvatar`
   */
  icon?: ReactNode;
  isDisabled?: boolean;
  /**
   * Adds/swaps for loading spinner & disables
   */
  isLoading?: boolean;
  /**
   * Changes visual of button to indicate it's currently selected
   */
  isSelected?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  /**
   * Adds a notification indicator for displaying the quantity provided
   */
  quantity?: number;
}

export const OuiFacetButton: FunctionComponent<OuiFacetButtonProps> = ({
  children,
  className,
  icon,
  isDisabled = false,
  isLoading = false,
  isSelected = false,
  quantity,
  buttonRef,
  ...rest
}) => {
  // If in the loading state, force disabled to true
  isDisabled = isLoading ? true : isDisabled;

  const classes = classNames(
    'ouiFacetButton',
    {
      'ouiFacetButton--isSelected': isSelected,
      'ouiFacetButton--unSelected': !isSelected,
    },
    className
  );

  // Add quantity number if provided or loading indicator
  let buttonQuantity: ReactElement;

  if (isLoading) {
    buttonQuantity = (
      <OuiLoadingSpinner className="ouiFacetButton__spinner" size="m" />
    );
  } else if (typeof quantity === 'number') {
    buttonQuantity = (
      <OuiNotificationBadge
        className="ouiFacetButton__quantity"
        size="m"
        color={!isSelected || isDisabled ? 'subdued' : 'accent'}>
        {quantity}
      </OuiNotificationBadge>
    );
  }

  // Add an icon to the button if one exists.
  let buttonIcon: ReactElement;

  if (React.isValidElement<{ className?: string }>(icon)) {
    buttonIcon = React.cloneElement(icon, {
      className: classNames(icon.props.className, 'ouiFacetButton__icon'),
    });
  }

  return (
    <OuiInnerText>
      {(ref, innerText) => (
        <button
          className={classes}
          disabled={isDisabled}
          type="button"
          ref={buttonRef}
          title={rest['aria-label'] || innerText}
          {...rest}>
          <span className="ouiFacetButton__content">
            {buttonIcon}
            <span
              className="ouiFacetButton__text"
              data-text={innerText}
              ref={ref}>
              {children}
            </span>
            {buttonQuantity}
          </span>
        </button>
      )}
    </OuiInnerText>
  );
};
