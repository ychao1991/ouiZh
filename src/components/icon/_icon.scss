/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

.ouiIcon {
  flex-shrink: 0; // Ensures it never scales down below its intended size
  display: inline-block;
  vertical-align: middle;
  fill: currentColor;
  transform: translate(0, 0); // Hack to fix Firefox "softness"

  &:focus {
    opacity: 1; // We often hide icons on hover. Make sure they appear on focus.
    background: $ouiFocusBackgroundColor;
  }
}

// App icons are two-toned so get an extra class. This provides the base color.
.ouiIcon--app {
  fill: $ouiTextColor;

  // This provides the default secondary color
  .ouiIcon__fillSecondary {
    fill: makeGraphicContrastColor($ouiColorSuccess);
  }
}

.ouiIcon-isLoading {
  opacity: $ouiIconLoadingOpacity;
  background-color: currentColor;
  border-radius: $ouiBorderRadius;
}

.ouiIcon-isLoaded {
  animation: ouiIconLoading $ouiAnimSpeedNormal ease-in 0s 1 forwards;
}

// COLORS

@each $name, $color in $ouiIconColors {
  .ouiIcon--#{$name} {
    // Use color attribute so currentColor can be used in more properties
    color: $color;
  }
}

// App icon special cases for fully mono
.ouiIcon--text,
.ouiIcon--subdued,
.ouiIcon--primary,
.ouiIcon--customColor {
  &,
  .ouiIcon__fillSecondary {
    fill: currentColor;
  }
}

// Use the negative fill on SVG elements that need to flip portions
// of the icon to light and dark compatible modes. You can see this
// in use on the Elastic logo icons.
.ouiIcon__fillNegative {
  fill: $ouiColorDarkestShade;
}

// SIZES

@each $name, $size in $ouiIconSizes {
  .ouiIcon--#{$name} {
    @include size($size);
  }
}

@keyframes ouiIconLoading {
  0% {
    opacity: $ouiIconLoadingOpacity;
  }

  100% {
    opacity: 1;
  }
}
