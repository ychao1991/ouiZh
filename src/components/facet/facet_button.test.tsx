/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { OuiFacetButton } from './facet_button';
import { OuiIcon } from '../icon';

describe('OuiFacetButton', () => {
  test('is rendered', () => {
    const component = render(
      <OuiFacetButton {...requiredProps}>Content</OuiFacetButton>
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('isDisabled', () => {
      it('is rendered', () => {
        const component = render(
          <OuiFacetButton isDisabled>Content</OuiFacetButton>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('isLoading', () => {
      it('is rendered', () => {
        const component = render(
          <OuiFacetButton isLoading>Content</OuiFacetButton>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('isSelected', () => {
      it('is rendered', () => {
        const component = render(
          <OuiFacetButton isSelected>Content</OuiFacetButton>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('quantity', () => {
      it('is rendered', () => {
        const component = render(
          <OuiFacetButton quantity={60}>Content</OuiFacetButton>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('icon', () => {
      it('is rendered', () => {
        const component = render(
          <OuiFacetButton icon={<OuiIcon type="dot" />}>Content</OuiFacetButton>
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick', () => {
        const handler = jest.fn();
        const component = mount(
          <OuiFacetButton onClick={handler}>Content</OuiFacetButton>
        );
        component.find('button').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });
    });
  });
});
