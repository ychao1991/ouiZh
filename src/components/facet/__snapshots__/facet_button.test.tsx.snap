// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiFacetButton is rendered 1`] = `
<button
  aria-label="aria-label"
  class="ouiFacetButton ouiFacetButton--unSelected testClass1 testClass2"
  data-test-subj="test subject string"
  title="aria-label"
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`OuiFacetButton props icon is rendered 1`] = `
<button
  class="ouiFacetButton ouiFacetButton--unSelected"
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__icon"
      data-ouiicon-type="dot"
    />
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`OuiFacetButton props isDisabled is rendered 1`] = `
<button
  class="ouiFacetButton ouiFacetButton--unSelected"
  disabled=""
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`OuiFacetButton props isLoading is rendered 1`] = `
<button
  class="ouiFacetButton ouiFacetButton--unSelected"
  disabled=""
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
    <span
      class="ouiLoadingSpinner ouiLoadingSpinner--medium ouiFacetButton__spinner"
    />
  </span>
</button>
`;

exports[`OuiFacetButton props isSelected is rendered 1`] = `
<button
  class="ouiFacetButton ouiFacetButton--isSelected"
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`OuiFacetButton props quantity is rendered 1`] = `
<button
  class="ouiFacetButton ouiFacetButton--unSelected"
  type="button"
>
  <span
    class="ouiFacetButton__content"
  >
    <span
      class="ouiFacetButton__text"
    >
      Content
    </span>
    <span
      class="ouiNotificationBadge ouiNotificationBadge--medium ouiNotificationBadge--subdued ouiFacetButton__quantity"
    >
      60
    </span>
  </span>
</button>
`;
